.github/workflows/cli.yml

```diff
@@ -70,6 +72,7 @@ jobs:
         with:
           platform: linux
           cache-key: ${{ runner.os }}-${{ runner.arch }}-cargo-dev-${{ hashFiles('cli/Cargo.lock') }}
+          restore-key: ${{ runner.os }}-${{ runner.arch }}-cargo-dev
```

Comment:
You're never building anything for that `restore-key`, no?

Reply:
The restore-key looks for any cache with the given prefix, so this restore-key will match an older version of the cache above.

I am actually wondering if the `restore-key`s are doing more harm than good, but that's something I want to play with in a future PR (if I get time)

Reply:
Ah right always forget it's a prefix. :D The only problem I see is that there won't be any cleanup, old artifacts will accumulate in the cache. Now how big of an issue it actually is, no idea. For the cli, it's probably fine to just try it out and see.

Reply:
> The only problem I see is that there won't be any cleanup, old artifacts will accumulate in the cache.

Yeah, that's exactly what I meant by "doing more harm than good".  But if we want to ditch them it'd be good to make that change in isolation so it's a bit clearer whether it is or isn't helping

Reply:
But... you're adding that change right now currently, no? What's there to change in later PR?